src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -201,8 +194,12 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                     routeValues["area"] = Area;
                 }
 
-                TagBuilder tagBuilder;
-                if (Route == null)
+                TagBuilder tagBuilder = null;
+                if (Action == null && Controller == null && Route == null && _routeValues == null && Fragment == null && Area == null)
```

Comment:
Yes, that could happen if a user did `<form asp-all-route-data="new Dictionary<string, string>()">`. In that case the user is being explicit and providing attributes; we'd want to follow the normal FormTagHelper rendering path.